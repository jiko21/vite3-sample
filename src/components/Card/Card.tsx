import { css } from '@emotion/react';
import React from 'react';

const style = css`
  background-color: #eeeeee;
  border-radius: 4px;
  padding: 8px;
`;

export type CardProps = {
  children: React.ReactNode
}

export const Card: React.FC<CardProps> = ({ children }) => <div css={style}>{children}</div>;
