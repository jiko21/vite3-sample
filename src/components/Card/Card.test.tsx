import { afterEach, describe, expect, it } from 'vitest';
import { Card } from './Card';
import { cleanup, render, screen } from '@testing-library/react';

describe('components/Card', () => {
  afterEach(cleanup);

  it('correctly render', () => {
    const { container } = render(<Card>Test</Card>);
    expect(screen.getByText('Test')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
