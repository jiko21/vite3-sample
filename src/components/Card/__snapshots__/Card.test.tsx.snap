// Vitest Snapshot v1

exports[`components/Card > correctly render 1`] = `
<div>
  <div
    class="css-1b7m767-Card"
  >
    Test
  </div>
</div>
`;
